benchmarks/vmt/grid_strided_vmt_main.cc

```diff
@@ -0,0 +1,396 @@
+        state.SetIterationTime(elapsed_seconds.count() -
+                               latency_measure->overhead_seconds);
+      } break;
+      case LatencyMeasureMode::kSystemSubmit: {
+        state.SetIterationTime(elapsed_seconds.count());
+      } break;
+      case LatencyMeasureMode::kGpuTimestamp: {
+        BM_CHECK_OK_AND_ASSIGN(
+            double timestamp_seconds,
+            query_pool->CalculateElapsedSecondsBetween(0, 1));
+        state.SetIterationTime(timestamp_seconds);
+      } break;
+    }
+
+    BM_CHECK_OK(cmdbuf->Reset());
+  }
+
+  double numOperation = double(N) * double(K) * 2.;
+  state.counters["Ops"] =
+      ::benchmark::Counter(numOperation,
+                           ::benchmark::Counter::kIsIterationInvariant |
+                               ::benchmark::Counter::kIsRate,
+                           ::benchmark::Counter::kIs1000);
+
+  // Reset the command pool to release all command buffers in the benchmarking
+  // loop to avoid draining GPU resources.
+  BM_CHECK_OK(device->ResetCommandPool());
+}
+
+// Returns true iff |a| is a multiple of |b|.
+static bool isMultipleOf(int a, int b) { return a >= b && a % b == 0; }
+
+namespace uvkc::benchmark {
+
+absl::StatusOr<std::unique_ptr<VulkanContext>> CreateVulkanContext() {
+  return CreateDefaultVulkanContext(kBenchmarkName);
+}
+
+bool RegisterVulkanOverheadBenchmark(
+    const vulkan::Driver::PhysicalDeviceInfo &physical_device,
+    vulkan::Device *device, double *overhead_seconds) {
+  return false;
+}
+
+void RegisterVulkanBenchmarks(
+    const vulkan::Driver::PhysicalDeviceInfo &physical_device,
+    vulkan::Device *device, const LatencyMeasure *latency_measure) {
+  const char *gpu_name = physical_device.v10_properties.deviceName;
+
+  const int N = 4096;
+  const int K = 4096;
+
+  for (const ShaderCode &shader : kShaderCodeCases) {
+    std::string vecmat_size = absl::StrCat(N, "x", K);
+    std::string tiling_scheme = absl::StrCat(shader.N0, "x", shader.K0);
+    BM_CHECK(isMultipleOf(N, shader.N0))
+        << "Incompatible tiling scheme: " << tiling_scheme;
+    BM_CHECK(isMultipleOf(K, shader.K0))
+        << "Incompatible tiling scheme: " << tiling_scheme;
+    BM_CHECK(isMultipleOf(shader.K0, 4))
+        << "Incompatible tiling scheme: " << tiling_scheme;
+
+    std::string workgroup_size =
+        absl::StrCat(shader.wg_size_x, "x", shader.wg_size_y, "x1");
+    std::string type_info = absl::StrCat(GetName(shader.input_type), "->",
+                                         GetName(shader.output_type));
+    std::string test_name = absl::StrCat(
+        gpu_name, "/vmt[", vecmat_size, "]/", type_info, "/", shader.name,
+        "/Workgroup", "_", shader.B0, "x", "[", workgroup_size, "]");
+    ::benchmark::RegisterBenchmark(test_name.c_str(), Vmt, device,
+                                   latency_measure, shader, N, K)
+        ->UseManualTime()
+        ->Unit(::benchmark::kMicrosecond);
```

Comment:
Can we have one `main.cc` file for both variants? This is a lot of that is hard to tell apart without running under `diff`. Or alternatively, maybe move common code into a header/library?